### ISSUE TITLE
Unable to read files generated by rexcel (rjava) plus suggested code fix

### ISSUE THREAD

**Issue by ChrisTantillo** [ExcelDataReader/ExcelDataReader#57]
Excel files generated by the statistical program R xlsx package (http://cran.r-project.org/web/packages/xlsx/xlsx.pdf) are incorrectly read by ExcelDataReader.  Only 1 column is read.

The same file reads correctly into Excel.  Doing a save-as from Excel can then be correctly opened by ExcelDataReader.

After investigation of the worksheet1.xml file in a debugger, it appears that the Dimension property of this file contained only "A1" and not "A1:S1206" like the value of Dimension when the file is saved by Excel.

As a result, ExcelDataReader thinks the number of columns is 0 (LastCol = 1) and only the first column of the worksheet is read.

A solution is suggested, which is to replace the single "Break;" statement in ExcelOpenXmlReader.cs, line ~130, by the following:

  // If LastCol is 1 then number of columns probably is wrong (although it might be just 1 column).
                    // Set dimension to null so rows and cols are calculated below.
                    if (sheet.Dimension.LastCol == 1)
                        sheet.Dimension = null;
                    else
                        break;

Basically, if LastCol == 1 then it probably did not read in the number of columns correctly so setting sheet.Dimension to null causes sheet.Dimension to be created further down in the code where the rows and columns are actually counted rather than determined from the Dimension element of the xml file.

**Comment by ChrisTantillo** [comment]
Screenshot of fix

![determiningdimension](https://cloud.githubusercontent.com/assets/8625764/6599746/e1ab85e2-c80b-11e4-9c28-bf2e1f9af389.PNG)

**Comment by ChrisTantillo** [comment]
I have an example file generated by R for testing but cannot figure out how to attach that to this ticket.  Feel free to contact me and I will email it if you want.

Thanks.

**Comment by Ian1971** [comment]
Thanks.

On Wed, Mar 11, 2015 at 3:33 PM, ChrisTantillo <notifications@github.com>
wrote:

> I have an example file generated by R for testing but cannot figure out
> how to attach that to this ticket. Feel free to contact me and I will email
> it if you want.
>
> Thanks.
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/ExcelDataReader/ExcelDataReader/issues/57#issuecomment-78287052>
> .
>



-- 

Ian Cox
C.T.O.

System Sync
m:            Matrix Studios, 91 Peterborough Road, London, SW6 3BU
w:            http://www.systemsyncsolutions.com
<http://www.releasemobile.com/>

Registered in England as systemsync solutions ltd.
Number: 9326325 VAT: 202729923

**Comment by andersnm** [comment]
This is the same as #142, and fixed in #144
